Providers simple metrics

cdsap opened the pull request:
Providers for simple metrics needed to be unify. 
After issues like #158 this is a starter point to remove the duplicated code and have the same simple metrics applied for all the different publishers.

MyDogTom commented:
As I understand, `ExecutionReport.flattenBuildEnv()` and `ValuesProvider.get()` serve the same purpose and just process different data. Based on that I have several questions:
* Should be there only one mechanism? Either only `ExecutionReport.flattenBuildEnv`, `ExecutionReport.flattenBuildMetrics`, `ExecutionReport.flattenTaskData` OR `DefaultBuildEnvProvider`, `DefaultBuildMetricsProvider`,`DefaultTaskDataProvider`.
* I see that `flattenBuildMetrics` is used only in `InfluxDbPublisher` and `PushGatewayFormatter` while `DefaultBuildMetricsProvider` is used also in `ElasticSearchPublisher` and `RethinkDbPublisher` is this a bug?
* Fields  `rootProject` and `rootProject` are duplicated between `ExecutionReport.flattenBuildEnd` and `DefaultBuildMetricsProvider`
* Shouldn't `DefaultBuildMetricsProvider` also have `.filter { (_, v) -> v != "undefined" }` in the end?

cdsap commented:
thank you very much for the comment, I addressed your comments in this commit:
https://github.com/cdsap/Talaiot/pull/160/commits/ddc5d61e70ee4c57f959363bdc4df396c001a2ff

> Should be there only one mechanism? Either only ExecutionReport.flattenBuildEnv, ExecutionReport.flattenBuildMetrics, ExecutionReport.flattenTaskData OR DefaultBuildEnvProvider, DefaultBuildMetricsProvider,DefaultTaskDataProvider.

Yep, you're right, just one source. Reworked the execution report and now only the `DefaultBuildMetricsProvider` is providing the data.

> I see that flattenBuildMetrics is used only in InfluxDbPublisher and PushGatewayFormatter while DefaultBuildMetricsProvider is used also in ElasticSearchPublisher and RethinkDbPublisher is this a bug?

Fixed, because now only depends on `DefaultBuildMetricsProvider `

> Fields rootProject and rootProject are duplicated between ExecutionReport.flattenBuildEnd and DefaultBuildMetricsProvider

Fixed 

> Shouldn't DefaultBuildMetricsProvider also have .filter { (_, v) -> v != "undefined" } in the end?

Fixed

I apply same unification in `DefaulTaskMetricsProvider`

Thanks again

MyDogTom commented:
Thank you! I have no other suggestions.
